pastas_plugins/pest/solver.py

```diff
@@ -0,0 +1,1159 @@
+        """
+        Initialize the PEST_HP solver.
+
+        Parameters
+        ----------
+        exe_name : Union[str, Path], optional
+            The name or path to the PEST_HP executable. Default is "pest_hp".
+        exe_agent : Union[str, Path], optional
+            The name or path to the agent_HP executable. Default is "agent_hp".
+        model_ws : Union[str, Path], optional
+            The model workspace directory for Pastas files. Default is "model".
+        temp_ws : Union[str, Path], optional
+            The template workspace directory for PEST files. Default is "temp".
+        noptmax : int, optional
+            The maximum number of optimization iterations. Default is 0.
+        control_data : Optional[dict], optional
+            Control data for the PEST solver. Default is None.
+        pcov : Optional[DataFrame], optional
+            The parameter covariance matrix. Default is None.
+        nfev : Optional[int], optional
+            The number of function evaluations. Default is None.
+        port_number : int, optional
+            The port number for communication. Default is 4004.
+        **kwargs : dict
+            Additional keyword arguments passed to the PestSolver.
+
+        Returns
+        -------
+        None
+        """
+        PestSolver.__init__(
+            self,
+            exe_name=exe_name,
+            model_ws=model_ws,
+            temp_ws=temp_ws,
+            pcov=pcov,
+            nfev=nfev,
+            long_names=False,
+            noptmax=noptmax,
+            control_data=control_data,
+            **kwargs,
+        )
+        self.exe_agent = Path(exe_agent)
+        self.port_number = port_number
+        self.computername = get_computername()
+        copy_file(self.exe_agent, self.temp_ws)  # copy agent executable
+
+    def solve(
+        self, silent: bool = False, **kwargs
+    ) -> Tuple[bool, np.ndarray, np.ndarray]:
+        """
+        Solve the optimization problem using the pest_hp solver.
+
+        This method sets up the model and necessary files, runs the solver, and
+        retrieves the optimal parameters and the objective function value.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments for the solver.
+
+        Returns
+        -------
+        success : bool
+            Indicates whether the solver ran successfully.
+        optimal : np.ndarray
+            The optimal parameters obtained from the solver.
+        stderr : np.ndarray
+            The standard errors of the optimal parameters.
+        """
+        self.setup_model()
+        self.setup_files(version=1)
+        # start consecutive thread for pest_hp and agent_hp excutable
+        threads = [
+            Thread(target=self.run, args=(f" /h :{self.port_number}", silent)),
+            Thread(target=self.run_agent, args=(silent,)),
+        ]
+        for t in threads:
+            t.start()
+            sleep(1.0)
```

Comment:
sleep this long? Or is it not bothersome to wait 1 second?

Reply:
It is a bit bothersome, but the time is needed to boot up the first executable. I tried shorter times but those were too fast.